importer/src/main/java/org/cbio/portal/pipelines/importer/config/tasklet/CancerStudyTasklet.java

```diff
@@ -0,0 +1,217 @@
+import org.springframework.jdbc.core.namedparam.*;
+
+/**
+ * Tasklet to import cancer study from meta_study.txt
+ * @author ochoaa
+ */
+@JobScope
+public class CancerStudyTasklet implements Tasklet {
+
+    @Value("#{jobParameters[stagingDirectory]}")
+    private String stagingDirectory;
+
+    @Autowired
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+    
+    private static final Log LOG = LogFactory.getLog(CancerStudyTasklet.class);
+    
+    @Override
+    public RepeatStatus execute(StepContribution stepContribution, ChunkContext chunkContext) throws Exception {
+        String metaFilename = stagingDirectory +  "meta_study.txt";
+        CancerStudy cancerStudy = loadCancerStudy(metaFilename);
+        
+        if (cancerStudy == null) {
+            return RepeatStatus.FINISHED;
+        }
+        
+        CancerStudy existingStudy = getCancerStudy(cancerStudy.getCANCER_STUDY_IDENTIFIER());
+        if (existingStudy != null) {
+            LOG.warn("Cancer study found with matching cancer study id: " + existingStudy.getCANCER_STUDY_IDENTIFIER());
+            deleteCancerStudy(existingStudy.getCANCER_STUDY_ID());
+        }
+
+        LOG.info("Importing cancer study: " + cancerStudy.getCANCER_STUDY_IDENTIFIER());
+        int assignedId = addCancerStudy(cancerStudy);
+        if (assignedId != -1) {
+            // update cancer study id and add cancer study to execution context
+            cancerStudy.setCANCER_STUDY_ID(assignedId);
+            chunkContext.getStepContext().getStepExecution().getJobExecution().getExecutionContext().put("cancerStudy", cancerStudy);
+        }
+        return RepeatStatus.FINISHED;
+    }
+
+    /**
+     * Load cancer study from meta_study.txt
+     * @param metaFilename
+     * @return
+     * @throws IOException 
+     */
+    private CancerStudy loadCancerStudy(String metaFilename) throws IOException {
+        File metaStudy = new File(metaFilename);
+        Properties properties = new Properties();
+        properties.load(new FileInputStream(metaStudy));
+
+        CancerStudy newCancerStudy = null;
+        try {
+            newCancerStudy = new CancerStudy(properties);
+        }
+        catch (NullPointerException ex) {}
+        
+        if (newCancerStudy == null) {
+            LOG.error("Error loading: " + metaFilename);
+            return null;
+        }
+        
+        return newCancerStudy;
+    }
+    
+    /**
+     * insert cancer study record into CANCER_STUDY
+     * @param cancerStudy
+     * @return
+     * @throws NoSuchMethodException
+     * @throws IllegalAccessException
+     * @throws IllegalArgumentException
+     * @throws InvocationTargetException 
+     */
+    private int addCancerStudy(CancerStudy cancerStudy) throws NoSuchMethodException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {   
+        String SQL = "INSERT INTO cancer_study " + 
```

Comment:
Spring 3 development ended before MyBatis 3 was officially released so the Ibatis item reader and writer are deprecated for now until the next release